src/query/plan/operator.cpp

```diff
@@ -3057,6 +3058,47 @@ void EdgeUniquenessFilter::EdgeUniquenessFilterCursor::Shutdown() { input_cursor
 
 void EdgeUniquenessFilter::EdgeUniquenessFilterCursor::Reset() { input_cursor_->Reset(); }
 
+EmptyResult::EmptyResult(const std::shared_ptr<LogicalOperator> &input, const std::vector<Symbol> &symbols)
+    : input_(input), symbols_(symbols) {}
+
+ACCEPT_WITH_INPUT(EmptyResult)
+
+std::vector<Symbol> EmptyResult::ModifiedSymbols(const SymbolTable &) const { return symbols_; }
+
+class EmptyResultCursor : public Cursor {
+ public:
+  EmptyResultCursor(const EmptyResult &self, utils::MemoryResource *mem)
+      : input_cursor_(self.input_->MakeCursor(mem)) {}
+
+  bool Pull(Frame &frame, ExecutionContext &context) override {
+    SCOPED_PROFILE_OP("EmptyResult");
+
+    // cache all the input
```

Comment:
This is not yet implemented and plz add a unit. Are you also planning to change the planner in this PR?

Reply:
Yes, planner will be changed in this PR.